integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/BrushTest.kt

```diff
@@ -89,28 +91,32 @@ fun CustomBrushTest() {
                     repeatMode = RepeatMode.Reverse
                 )
         )
-
+    lateinit var brushState: State<Brush>
     // Android T introduces AGSL and RuntimeShader. Robolectric (part of our test infrastructure)
     // only supports software rendering.
-    val brush: () -> Brush =
-        if (
-            Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU &&
-                "robolectric" != Build.FINGERPRINT
-        ) {
-            val shader = RuntimeShader(CUSTOM_SHADER)
-            val shaderBrush = SizingShaderBrush(shader)
-            // The kotlin compiler seems to get confused without semicolons on these statements.
-            shader.setFloatUniform("iTime", 0.0f);
-
-            {
-                // Only sample the state in the generator function; this means that we avoid
-                // recomposition and only do the redraw phase.
-                shader.setFloatUniform("iTime", movingValue.value)
-                shaderBrush
+    if (
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU && "robolectric" != Build.FINGERPRINT
+    ) {
+        val shader = RuntimeShader(CUSTOM_SHADER)
+        val shaderBrush = SizingShaderBrush(shader)
+        // The kotlin compiler seems to get confused without semicolons on these statements.
+        shader.setFloatUniform("iTime", 0.0f)
+        brushState =
+            object : State<Brush> {
+                override val value: Brush
+                    get() =
+                        shaderBrush.also {
+                            // Only sample the state in the getter function; this means that we
+                            // avoid recomposition and only do the redraw phase.
+                            shader.setFloatUniform("iTime", movingValue.value)
+                        }
             }
-        } else {
-            { SolidColor(Color.Blue) }
-        }
-
-    CustomBrushTestDoc.Main(fill = brush)
+    } else {
+        brushState =
+            object : State<Brush> {
+                override val value: Brush
+                    get() = SolidColor(Color.Blue)
+            }
+    }
+    CustomBrushTestDoc.Main(fill = brushState)
```

Comment:
I guess this is existing code, but it seems rather complicated for an example. Is this a common pattern for State objects?

Reply:
Oh I didn't know.
It can create non mutable state just using `remember { SolidColor(Color.Blue) }`